mangrove-core/how-to-guides/DirectHowTo.md

```diff
@@ -0,0 +1,307 @@
+
+```solidity
+pragma solidity ^0.8.10;
+
+pragma abicoder v2;
+
+import "mgv_src/strategies/offer_maker/abstract/Direct.sol";
+import "mgv_src/strategies/routers/AbstractRouter.sol";
+
+contract OfferMaker is Direct {
+
+  constructor(IMangrove mgv, AbstractRouter router_, address deployer) Direct(mgv, router_, 30_000) {
+    // stores total gas requirement of this strat (depends on router gas requirements)
+    // if contract is deployed with static address, then one must set admin to something else than msg.sender
+    if (deployer != msg.sender) {
+      setAdmin(deployer);
+    }
+  }
+}
+```
+
+Technically Direct does not require anything else. But since the newOffer function of Direct is an internal function, deploying the contract as-is, would not allow anyone to post an offer. Because of this we want to add one thing. We want to implement the IMakerLogic, which only says that the contract has to have a newOffer function with the correct parameters. You could choose to not use this interface, but it is a nice help, that enforces that the offer maker gives all the relevant information for posting a new offer. Since we only want the admin of the contract to be able to post offers, we add the modifier `onlyAdmin` to the function. This is a modifier Direct can use, because it is a `AccessControlled` contract.
+
+When this is added, then the contract is ready to be [deployed](HowToDeploy.md). The contract can now post new offers, update offers and retract offers, using all the default behavior from a Direct contract.
+
+```solidity
+pragma solidity ^0.8.10;
+
+pragma abicoder v2;
+
+import "mgv_src/strategies/offer_maker/abstract/Direct.sol";
+import "mgv_src/strategies/routers/AbstractRouter.sol";
+import "mgv_src/strategies/interfaces/IMakerLogic.sol";
+
+contract OfferMaker is IMakerLogic, Direct {
+
+  constructor(IMangrove mgv, AbstractRouter router_, address deployer) Direct(mgv, router_, 30_000) {
+    // stores total gas requirement of this strat (depends on router gas requirements)
+    // if contract is deployed with static address, then one must set admin to something else than msg.sender
+    if (deployer != msg.sender) {
+      setAdmin(deployer);
+    }
+  }
+
+  function newOffer(
+    IERC20 outbound_tkn,
+    IERC20 inbound_tkn,
+    uint wants,
+    uint gives,
+    uint gasreq,
+    uint gasprice,
+    uint pivotId
+  ) public payable override onlyAdmin returns (uint offerId) {
+    offerId = _newOffer(
+      OfferArgs({
+        outbound_tkn: outbound_tkn,
+        inbound_tkn: inbound_tkn,
+        wants: wants,
+        gives: gives,
+        gasreq: gasreq,
+        gasprice: gasprice,
+        pivotId: pivotId,
+        fund: msg.value,
+        noRevert: false,
+        caller: msg.sender
+      })
+    );
+  }
+}
+```
+
+## Posting ghost liquidity with Direct
```

Comment:
I think discussing here the script to deploy the contract (and activate it) would be welcome.
The tutorial for ghost should come as a continuation later. And by the way, the name Ghost is a bit pejorative. Maybe we should use `LiquidityAmplifyer` or something like this

Reply:
Agreed, we could probably split it up a bit and use a better name. Not sure what to change it to though